src/models/popUp.js

```diff
@@ -194,10 +198,36 @@ const popUpModel = {
             // TODO: try making a 1D growth comparator
 
             // TODO: position = absolute or fixed
+            function absolutePosition(fnode) {
+                return ((fnode.element.style.position == "absolute") ? ONEISH : ZEROISH) ** coeffAbsolutePosition;
+            }
             // TODO: align items: centered
 
+
+
             // TODO: try z-index again
+            
+            function getMaxZIndex() {
+                console.log('should only go into this once???');
+                 return Array.from(document.querySelectorAll('body *'))
+                       .map(a => parseFloat(window.getComputedStyle(a).zIndex))
+                       .filter(a => !isNaN(a))
+                       .sort()
+                       .pop();
+            }
 
+            function inheritedZIndex(fnode){
+                //maybe use a 'note' in the root element. there's a get root element util function
+                if(!window.maxZ){
+                    window.maxZ = getMaxZIndex();
+                }
+                for (const e of ancestors(fnode.element)){
+                    if (window.getComputedStyle(e).zIndex !== 'auto'){
```

Comment:
Just for educational purposes, it wouldn't go amiss to stick the result of getComputedStyle(e).zIndex into a var and thus make the DOM calls only once (and shorten the code while you're at it). The DOM calls are something like 60-80% of a typical Fathom ruleset's runtime.